exchanges/gateio/gateio_wrapper.go

```diff
@@ -2578,3 +2559,215 @@ func (g *Gateio) GetCurrencyTradeURL(_ context.Context, a asset.Item, cp currenc
+			Status:               status,
+			ImmediateOrCancel:    responses[x].TimeInForce == iocTIF,
+			FillOrKill:           responses[x].TimeInForce == fokTIF,
+			PostOnly:             responses[x].TimeInForce == pocTIF,
+			Cost:                 cost,
+			Purchased:            purchased,
+			Fee:                  responses[x].Fee.Float64(),
+			FeeAsset:             responses[x].FeeCurrency,
+		})
+	}
+	return out, nil
+}
+
+// deriveFuturesWebsocketOrderResponses returns the order submission responses for futures
+func (g *Gateio) deriveFuturesWebsocketOrderResponses(responses []WebsocketFuturesOrderResponse) ([]*order.SubmitResponse, error) {
+	if len(responses) == 0 {
+		return nil, common.ErrNoResponse
+	}
+
+	out := make([]*order.SubmitResponse, 0, len(responses))
+	for x := range responses {
+		status := order.Open
+		if responses[x].FinishAs != "" && responses[x].FinishAs != statusOpen {
+			var err error
+			status, err = order.StringToOrderStatus(responses[x].FinishAs)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		oType := order.Market
+		if responses[x].Price != 0 {
+			oType = order.Limit
+		}
+
+		side := order.Long
+		if responses[x].Size < 0 {
+			side = order.Short
+		}
+
+		var clientOrderID string
+		if responses[x].Text != "" && strings.HasPrefix(responses[x].Text, "t-") {
+			clientOrderID = responses[x].Text
+		}
+
+		out = append(out, &order.SubmitResponse{
+			Exchange:             g.Name,
+			OrderID:              strconv.FormatInt(responses[x].ID, 10),
+			AssetType:            asset.Futures,
+			Pair:                 responses[x].Contract,
+			ClientOrderID:        clientOrderID,
+			Date:                 responses[x].CreateTime.Time(),
+			LastUpdated:          responses[x].UpdateTime.Time(),
+			RemainingAmount:      math.Abs(responses[x].Left),
+			Amount:               math.Abs(responses[x].Size),
+			Price:                responses[x].Price.Float64(),
+			AverageExecutedPrice: responses[x].FillPrice.Float64(),
+			Type:                 oType,
+			Side:                 side,
+			Status:               status,
+			ImmediateOrCancel:    responses[x].TimeInForce == iocTIF,
+			FillOrKill:           responses[x].TimeInForce == fokTIF,
+			PostOnly:             responses[x].TimeInForce == pocTIF,
+			ReduceOnly:           responses[x].IsReduceOnly,
+		})
+	}
+	return out, nil
+}
+
+func (g *Gateio) getSpotOrderRequest(s *order.Submit) (*CreateOrderRequest, error) {
+	switch {
+	case s.Side.IsLong():
+		s.Side = order.Buy
+	case s.Side.IsShort():
+		s.Side = order.Sell
+	default:
+		return nil, order.ErrSideIsInvalid
+	}
+
+	timeInForce, err := getTimeInForce(s)
+	if err != nil {
+		return nil, err
+	}
+
+	return &CreateOrderRequest{
+		Side:    s.Side.Lower(),
+		Type:    s.Type.Lower(),
+		Account: g.assetTypeToString(s.AssetType),
+		// When doing spot market orders when purchasing base currency, the quote currency amount is used. When selling
+		// the base currency the base currency amount is used.
```

Comment:
This comment feels superfluous, because we're not doing anything related to that logic here.
```suggestion
```

Also / otherwise: I'm not sure what about this was market order related :)
Would also amount to just `Buy in Quote currency, Sell in Base`